### PR TITLE
Edit form

### PR DESCRIPTION
**Pull request by OleksiiPliekhov**

### CODE REVIEW

File: src/component/Sidebar.tsx [hunk 1]
```diff
@@ -26,7 +26,7 @@ function SideBar() {
             <CiSearch className="m-5 text-4xl hover:scale-110"/>
         </div>
         <div className="w-full h-1/6 border-t-2 border-white flex justify-center items-center">
-            <BsPersonCircle className="text-6xl"/>
+            <BsPersonCircle className="text-6xl" onClick={() => {navigate("/profile")}}/>
```
Comment (havriutkin) [inline src/component/Sidebar.tsx]: When you navigate to profile, you should pass email that you want to be fetched on Profile page

File: src/component/ProfileEditForm.tsx [hunk 2]
```diff
@@ -23,11 +25,28 @@ interface ProfileEditFormProps {
 
 function ProfileEditForm({ onClose, onSave }: ProfileEditFormProps){
     const { register, formState: {errors}, handleSubmit } = useForm<ProfileEditFormInput>();
+    const updateUser = useUserStore(state => state.updateUser);
 
-    const onSubmit: SubmitHandler<ProfileEditFormInput> = (data) => {
+    const onSubmit: SubmitHandler<ProfileEditFormInput> = async (data, e) => {
+        e?.preventDefault();
+        const userData: User = {
+            id: 0,
```
Comment (havriutkin) [inline src/component/ProfileEditForm.tsx:33]: You can omit id field

File: src/component/ProfileEditForm.tsx [hunk 3]
```diff
@@ -23,11 +25,28 @@ interface ProfileEditFormProps {
...
+            experience: data.experience,
+            bio: data.bio,
+            gitHubLink: data.github,
+            skills: [], // You might not have skills in the form data
+        };
         console.log(data);
-        onSave(data);
-    }
+        console.log(userData);
+        await updateUser(userData).then(() => {
```
Comment (havriutkin) [inline src/component/ProfileEditForm.tsx]: There is no point in using both await and then. If you use "then" - remove await.

File: src/store/user.ts [hunk 4]
```diff
@@ -4,7 +4,7 @@ import useAuthStore from "./auth";
 
 const ENDPOINT = import.meta.env.VITE_API_ENDPOINT;
 
-interface User {
+export interface User {
```
Comment (havriutkin) [inline src/store/user.ts]: This file is supposed to export only useAuthStore. If we want to export types, we should create model folder

File: src/store/user.ts [hunk 5]
```diff
@@ -57,7 +49,34 @@ const useUserStore = create<UserState & UserActions>((set) => ({
...
+        const token = useAuthStore.getState().token;
+
+        try {
+            const response = await axios.put(`${ENDPOINT}/user/${useAuthStore.getState().principal?.id}`, data, { // Use data directly
+                headers: {
+                    Authorization: `Bearer ${token}`
+                }
+            });
+            if (response.status === 200) {
+                fetchUser(response.data.email);
```
Comment (havriutkin) [inline src/store/user.ts]: Why do we need to call fetchUser here? We set user state on the next line

File: src/component/ProfileEditForm.tsx [hunk 6]
```diff
@@ -23,11 +25,28 @@ interface ProfileEditFormProps {
 
 function ProfileEditForm({ onClose, onSave }: ProfileEditFormProps){
     const { register, formState: {errors}, handleSubmit } = useForm<ProfileEditFormInput>();
+    const {user} = useUserStore();
```
Comment (havriutkin) [inline src/component/ProfileEditForm.tsx]: I believe you can just use `const { user, updateUser } = useUserStore();`

File: src/component/ProfileEditForm.tsx [hunk 7]
```diff
@@ -23,11 +25,28 @@ interface ProfileEditFormProps {
...
+    const updateUser = useUserStore(state => state.updateUser);
 
-    const onSubmit: SubmitHandler<ProfileEditFormInput> = (data) => {
-        console.log(data);
-        onSave(data);
-    }
+    const onSubmit: SubmitHandler<ProfileEditFormInput> = async (data, e) => {
+        e?.preventDefault();
+        const userData: User = {
+            id: 0,
```
Comment (havriutkin) [inline src/component/ProfileEditForm.tsx]: Still don't really like using 0 for id. Try to just use `user.id` that you have from store

**Comment by OleksiiPliekhov** [comment]
Now users` skills are visible on Profile Page

**Pull request merged**